xbmc/cores/VideoPlayer/DVDInputStreams/DVDStateSerializer.cpp

```diff
@@ -189,105 +54,68 @@ bool CDVDStateSerializer::DVDToXMLState( std::string &xmlstate, const dvd_state_
   return true;
 }
 
-bool CDVDStateSerializer::XMLToDVDState( dvd_state_t *state, const std::string &xmlstate )
+bool CDVDStateSerializer::XMLToDVDState(DVDState& state, const std::string& xmlstate)
 {
   CXBMCTinyXML xmlDoc;
 
   xmlDoc.Parse(xmlstate);
-
-  if( xmlDoc.Error() )
+  if (xmlDoc.Error())
     return false;
 
-  TiXmlHandle hRoot( xmlDoc.RootElement() );
-  if( strcmp( hRoot.Element()->Value(), "navstate" ) != 0 ) return false;
-
-  TiXmlElement *element = NULL;
-  TiXmlText *text = NULL;
-  int index = 0;
-
-  element = hRoot.FirstChildElement("registers").FirstChildElement("sprm").Element();
-  while( element )
+  TiXmlHandle hRoot(xmlDoc.RootElement());
+  if (!hRoot.Element() || !StringUtils::EqualsNoCase(hRoot.Element()->Value(), "navstate"))
   {
-    element->Attribute("index", &index);
-
-    text = TiXmlHandle( element ).FirstChildElement("value").FirstChild().Text();
-    if( text && index >= 0 && index < 24 )
-      sscanf(text->Value(), "0x%hx", &state->registers.SPRM[index]);
+    CLog::LogF(LOGERROR, "Failed to deserialize dvd state - failed to detect root element.");
+    return false;
+  }
 
-    element = element->NextSiblingElement("sprm");
+  auto version = hRoot.Element()->Attribute("version");
+  if (!version || !StringUtils::EqualsNoCase(version, std::to_string(DVDSTATESERIALIZER_VERSION)))
+  {
+    CLog::LogF(LOGERROR, "Failed to deserialize dvd state - wrong provided serializer version.");
+    return false;
   }
 
-  element = hRoot.FirstChildElement("registers").FirstChildElement("gprm").Element();
-  while( element )
+  for (const TiXmlElement* childElement = hRoot.Element()->FirstChildElement();
+       childElement != nullptr; childElement = childElement->NextSiblingElement())
```

Comment:
this works, but i'd keep the while loop. it's a bit easier to read (imo). you should still be able to initialize `const TiXmlElement* childElement = ...`